aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,486 @@
+Sample: 
+ 
+.. literalinclude::  forms/sample/final/ViewModels/RegisterViewModel.cs
+  :language: c#
+
+.. literalinclude::  forms/sample/final/Views/Demo/RegisterInput.cshtml
+  :language: HTML
+
+The code above generates the following HTML:
+
+.. code-block:: HTML
+
+    <form method="post" action="/Demo/RegisterInput">
+      Email:  
+      <input type="email" data-val="true" 
+             data-val-email="The Email Address field is not a valid e-mail address." 
+             data-val-required="The Email Address field is required." 
+             id="Email" name="Email" value="" /> <br>
+      Password: 
+      <input type="password" data-val="true" 
+             data-val-required="The Password field is required." 
+             id="Password" name="Password" /><br>
+      <button type="submit">Register</button>
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+  </form>
+
+The data annotations applied to the ``Email`` and ``Password`` properties generated meta data on the model in the form of `HTML5 <https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5>`__ ``data-val-*`` attributes (see :doc:`/mvc/models/validation`). (``data-val-`` specifies data validation.) The Input Tag Helper consumes the model metadata and produces HTML5-compatible attributes describing the validators to attach to the input fields. This provides unobtrusive HTML5 and `jQuery <https://jquery.com/>`__ validation. The unobtrusive attributes have the format ``data-val-rule="Error Message"``, where rule is the validation rule (such as required, email format, minimum string length, and maximum string length). If an error message is provided in the attribute, it is displayed as the value for the ``data-val-rule`` attribute. 
+
+Navigating child properties
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can also navigate to child properties of your view model. Consider a more complex model class that contains a child ``Address`` property.
+
+.. literalinclude::  forms/sample/final/ViewModels/AddressViewModel.cs
+  :language: c#
+  :lines: 5-16
+  :dedent: 3
+  :emphasize-lines: 1-4,11
+
+In the view, we bind to ``Address.AddressLine1``: 
+
+.. literalinclude::  forms/sample/final/Views/Demo/RegisterAddress.cshtml 
+  :language: HTML
+  :emphasize-lines: 6
+
+The following HTML is generated for `` Address.AddressLine1``:
+
+.. code-block:: HTML
+
+  <input type="text" id="Address_AddressLine1" name="Address.AddressLine1" value="" />
+  
+The Textarea Tag Helper
+-------------------------
+
+The `Textarea Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/TextAreaTagHelper/index.html>`__ tag helper is  similar to the Input Tag Helper.  
+
+- Generates the ``id`` and ``name`` attribute, and the data validation attributes from the model for a `<textarea> <http://www.w3.org/wiki/HTML/Elements/textarea>`__ element. 
+- HTML Helper alternative: ``Html.TextAreaFor``
+
+Consider the following form:
```

Comment:
- Stick with introducing the model or view model classes and _then_ the tag helper using it / them.
- Also prefer simpler "Sample" over the formal "Consider the ..."